### PR TITLE
Add utoipa annotation for /infra/{id}/pathfinding 

### LINKED ISSUE #6317: Impossible to create a route
### What happened?

When creating a route we need to search for compatible routes between two detectors/buffer stops.

Request `POST` `/infra/8/pathfinding/` fails with a `400`.

```json
{
  "status": 400,
  "type": "editoast:JsonError",
  "context": {
    "cause": "Json deserialize error: unknown field `track_section`, expected `track` or `position` at line 1 column 28"
  },
  "message": "Json deserialize error: unknown field `track_section`, expected `track` or `position` at line 1 column 28"
}
```

-----------

There is an error in the editoast openapi. Ideally we should make it use utoipa.

### What did you expect to happen?

We should see possible routes.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to editor
2. Try to create a route
3. When searching for compatible routes we have an infinite loader

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local & Dev

### OSRD version (top right corner `Account` button > `Informations`)

bf6da3d

### PR DESCRIPTION
**Pull request by flomonster**
refs https://github.com/osrd-project/osrd/issues/4580 https://github.com/osrd-project/osrd/issues/3625
close #6317

### CODE REVIEW

File: editoast/src/views/infra/pathfinding.rs [hunk 1]
```diff
@@ -66,16 +74,24 @@ struct QueryParam {
 }
 
 /// This endpoint search path between starting and ending track locations
-#[post("/pathfinding")]
+#[utoipa::path(
+    tag = "infra",
+    params(InfraIdParam),
```
Comment (leovalais) [inline editoast/src/views/infra/pathfinding.rs]:
```suggestion
    params(InfraIdParam, QueryParams),
```

File: editoast/src/views/infra/pathfinding.rs [hunk 2]
```diff
@@ -66,16 +74,24 @@ struct QueryParam {
 }
 
 /// This endpoint search path between starting and ending track locations
-#[post("/pathfinding")]
+#[utoipa::path(
+    tag = "infra",
```
Comment (leovalais) [inline editoast/src/views/infra/pathfinding.rs]:
```suggestion
    tag = "infra,pathfinding",
```

**Review by leovalais — APPROVED** [review]
LGTM, a few omissions:

**Review by Math-R — APPROVED** [review]
LTGM tested as you mention and it works !

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested

**Review by Tguisnet — APPROVED** [review]
Lgtm

**Pull request merged**